src/librustc_interface/interface.rs

```diff
@@ -109,7 +117,15 @@ where
         output_file: config.output_file,
         queries: Default::default(),
         crate_name: config.crate_name,
-    };
+        interp_user_fn: None,
+    }
+}
+
+pub fn run_compiler_in_existing_thread_pool<F, R>(config: Config, f: F) -> R
```

Comment:
Nit: `impl FnOnce(&Compiler) -> R` would be cleaner here imo.

Reply:
Yep. Pre-existing code essentially, but fair point.